modules/telemetry/telemetry.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+namespace VIPWorkflow\Modules\Telemetry;
+
+use Automattic\VIP\Telemetry\Tracks;
+use VIPWorkflow\Modules\CustomStatus;
+use VIPWorkflow\Modules\Shared\PHP\HelperUtilities;
+use WP_Post;
+use WP_Term;
+
+class Telemetry {
+	/**
+	 * Tracker instance
+	 *
+	 * @var Tracks
+	 */
+	protected static Tracks $tracks;
+
+	/**
+	 * Initialize the module and register event callbacks
+	 */
+	public static function init(): void {
+		self::$tracks = new Tracks( 'vip_workflow_', [ 'plugin_version' => VIP_WORKFLOW_VERSION ] );
+
+		// Custom Status events
+		add_action( 'transition_post_status', [ __CLASS__, 'record_custom_status_change' ], 10, 3 );
+		add_action( 'vw_add_custom_status', [ __CLASS__, 'record_add_custom_status' ], 10, 3 );
+		add_action( 'vw_delete_custom_status', [ __CLASS__, 'record_delete_custom_status' ], 10, 2 );
+		add_action( 'vw_update_custom_status', [ __CLASS__, 'record_update_custom_status' ], 10, 2 );
+
+		// Notification events
+		add_action( 'vw_notification_status_change', [ __CLASS__, 'record_notification_sent' ], 10, 3 );
+
+		// Settings events
+		add_action( 'vw_upgrade_version', [ __CLASS__, 'record_admin_update' ], 10, 2 );
+		add_action( 'vw_save_settings', [ __CLASS__, 'record_settings_update' ], 10, 2 );
+	}
+
+	// Custom Status events
+
+	/**
+	 * Record an event when a post's custom status changes
+	 *
+	 * @param string $new_status The new status
+	 * @param string $old_status The old status
+	 * @param WP_Post $post The post object
+	 */
+	public static function record_custom_status_change( string $new_status, string $old_status, WP_Post $post ): void {
+		if ( ! in_array( $post->post_type, HelperUtilities::get_supported_post_types() ) ) {
+			// This isn't a supported post type
+			return;
+		} elseif ( $old_status === $new_status ) {
+			// The status hasn't changed
+			return;
+		} elseif ( in_array( $new_status, [ 'inherit', 'auto-draft' ] ) ) {
+			// The status hasn't changed, or it moved into an auto-generated status
+			return;
+		}
+
+		$status_slugs = wp_list_pluck( CustomStatus::get_custom_statuses(), 'slug' );
+
+		if ( ! in_array( $new_status, $status_slugs, true ) && ! in_array( $old_status, $status_slugs, true ) ) {
```

Comment:
The logic in this function has been modified so that we'll still track if a post moves from `some-custom-status` to `publish`. If the prior or current status is custom, we track.